projects/plugins/jetpack/modules/shortcodes/vimeo.php

```diff
@@ -250,6 +259,7 @@ class_exists( 'Jetpack_AMP_Support' )
  * Callback to modify output of embedded Vimeo video using Jetpack's shortcode.
  *
  * @since 3.9
+ * @deprecated since x.x.x
```

Comment:
On top of that deprecation warning, I think you could add a warning to developers via `_deprecated_function`:
https://developer.wordpress.org/reference/functions/_deprecated_function/

https://github.com/Automattic/jetpack/blob/f0fa2c82dce0cb507b892645e27a9fafd8ab87e0/docs/coding-guidelines.md#L54-L55